### PR TITLE
feat: render sitemap from `sitemap.json`

### LINKED ISSUE #472: deliver or render sitemap.xml
**Description**
the pipeline should render a simple sitemap.xml if it has an autogenerated sitemap.json in the content bus.
it should use the correct canonical host for the hrefs.

### PR DESCRIPTION
**Pull request by dominique-pfister**
fix #472

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,13 +47,15 @@
     "@adobe/mdast-util-gridtables": "4.0.0",
     "@adobe/remark-gridtables": "3.0.0",
     "cookie": "0.6.0",
+    "dayjs": "1.11.10",
```
Comment (dominique-pfister) [inline package.json:50]: new dependency (which has no other dependency)

File: package.json [hunk 2]
```diff
@@ -47,13 +47,15 @@
...
     "github-slugger": "2.0.0",
     "hast-util-raw": "9.0.2",
     "hast-util-select": "6.0.2",
     "hast-util-to-html": "9.0.0",
     "hast-util-to-string": "3.0.0",
     "hastscript": "9.0.0",
     "jose": "5.2.1",
+    "lodash.escape": "4.0.1",
```
Comment (dominique-pfister) [inline package.json:58]: new dependency (which has no other dependency)

File: src/sitemap-pipe.js [hunk 3]
```diff
@@ -34,8 +76,6 @@ import { PipelineResponse } from './PipelineResponse.js';
 export async function sitemapPipe(state, req) {
   const { log } = state;
   state.type = 'sitemap';
-  // force loading from preview
-  state.partition = 'preview';
```
Comment (dominique-pfister) [inline src/sitemap-pipe.js:38]: removed that again, or we lose the information on the `origin` to use in generated `sitemap.xml`

File: package.json (hunk 1 above)
Comment (dominique-pfister) [reply package.json:50]: realized that we don't need *that* flexibility, but could use a hardcoded date representation (e.g. `YYYY-MM-DD`), so I can get rid of it

File: test/fixtures/content/config-all.json [hunk 4]
```diff
@@ -7,6 +7,9 @@
       "cdn": {
         "prod": {
           "route": "/express/"
+        },
+        "preview": {
+          "host": "$ref--$repo--$owner.my.page"
```
Comment (dominique-pfister) [inline test/fixtures/content/config-all.json:12]: added this to test preview host

File: src/sitemap-pipe.js [hunk 5]
```diff
@@ -20,6 +21,48 @@ import setCustomResponseHeaders from './steps/set-custom-response-headers.js';
...
+    log.info('failed to parse /sitemap.json', e);
+    throw new PipelineStatusError(404, `Failed to parse /sitemap.json: ${e.message}`);
+  }
+  const { data } = config;
+  if (!data || !Array.isArray(data)) {
+    throw new PipelineStatusError(404, 'Expected \'data\' array not found in /sitemap.json');
+  }
+  const host = partition === 'preview'
+    ? (previewHost || `${ref}--${repo}--${owner}.hlx.page`)
+    : (prodCDN || `${ref}--${repo}--${owner}.hlx.live`);
```
Comment (tripodsan) [inline src/sitemap-pipe.js:46]:
```suggestion
    : (prodCDN || liveHost || `${ref}--${repo}--${owner}.hlx.live`);
```

also use `state.liveHost` if available.

File: src/sitemap-pipe.js [hunk 6]
```diff
@@ -20,6 +21,48 @@ import setCustomResponseHeaders from './steps/set-custom-response-headers.js';
...
+    <loc>
+      https://${host}${escape(path)}
+    </loc>
+    <lastmod>${new Date(lastModified * 1000).toISOString().substring(0, 10)}</lastmod>
+  </url>`;
+  const xml = `<?xml version="1.0" encoding="utf-8"?>
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">
+${data.map((record) => loc(record)).join('\n')}
+</urlset>
+`;
```
Comment (tripodsan) [inline src/sitemap-pipe.js:57]: isn't it faster (and nicer to read) to create an array of all lines and then join at the end?

```
[
  '<?xml version="1.0" encoding="utf-8"?>',
  '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">',
  ...data..map((record) => loc(record)),
  '</urlset>'
].join('\n);
```

File: src/sitemap-pipe.js (hunk 3 above)
Comment (tripodsan) [reply src/sitemap-pipe.js:38]: but it important that the sitemap.xml and/or sitemap.json is read from preview

**Review by tripodsan — CHANGES_REQUESTED** [review]
- also respect `liveHost`
- consider generating the xml with array of lines
- ensure that sitemap.xml is always loaded from `preview` partition.

**Review by tripodsan — APPROVED** [review]

**Pull request merged**